projects/kit/components/select/select.component.ts

```diff
@@ -144,6 +145,12 @@ export class TuiSelectComponent<T>
         this.value = value || null;
     }
 
+    onClick(event: MouseEvent): void {
+        if (tuiRetargetedBoundaryCrossing(event)) {
+            event.stopImmediatePropagation();
```

Comment:
Maybe we should just handle this inside primitive textfield? Any reason against it?

Reply:
fixed